presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/MetadataResolver.java

```diff
@@ -87,4 +90,24 @@ default boolean isMaterializedView(QualifiedObjectName viewName)
     {
         return getMaterializedView(viewName).isPresent();
     }
+
+    /**
+     * Returns list of registered types.
+     */
+    List<Type> getTypes();
```

Comment:
Similar comment as functions. Do we see the value of returning a list of types? To me the most important two functionalities are:
- `Type getType(TypeSignature signature)` and
-  `boolean canCoerce(Type actualType, Type expectedType)`

Of course, it depends on how we design the analyzer.

Reply:
canCoerce seems useful. I think crux has its own logic to check coercion if I remember correctly. I can double check on this tomorrow.